lib/resty/upstream/healthcheck.lua

```diff
@@ -524,28 +533,33 @@ end
 
 function _M.spawn_checker(opts)
     local typ = opts.type
-    if not typ then
+    if typ == nil then
         return nil, "\"type\" option required"
     end
 
-    if typ ~= "http" then
-        return nil, "only \"http\" type is supported right now"
+    local host = opts.host
+    if host == nil then
+        host = nil
+    end
```

Comment:
seems these lines do nothing?

Reply:
Yes, you're right, I'll remove those lines.